internal/edge/stack.go

```diff
@@ -192,10 +198,12 @@ func (manager *StackManager) start() error {
 					continue
 				}
 
+				ctx := context.TODO()
```

Comment:
Why don't you set the mutex lock in the `default ` block here instead of everywhere else?

Right before

```
stack := manager.next()
```

Reply:
That is an option I considered, but I worry that it would be more error-prone, for example if some other code needs to call to one of these functions in the future. In addition, having a lock while we wait for the timer doesn't sound that good, it would leave less opportunity for other functions to acquire the lock, so I would have to acquire and release two times in that same loop anyway.

Also, adding a `defer` in the way you proposed would result in a deadlock, so we need to manually `Unlock()` before the `continue` and before the loop finishes,